docs/specs.md

```diff
@@ -87,181 +113,323 @@ can be considered as _a collection of text chunks_. How those chunks are
+   `data` and `context`
+ * The `data` element contains the contents of a single document cell
+ * The context elements have the usual _global_, _before_ and _after_ fields
+   **plus** one additional field:
+     - _column_: contains the column name and index for the cell in this chunk
+ * Note that here _before_ and _after_ refer to the previous & next cell,
+   which for the first/last cells in a row will be in another row
 
-* **id**: an arbitrary string that should be unique per document. Its mission is to make it easier later on to map detected PII instances to the chunk they are part of
-* **text**: a text section that contains the textual contents of the chunk. It will be a string containing UTF-8 raw text. It can contain newlines or blank lines, to be considered as part of the text structure, but no formatting or layout contents (it is assumed that exact formatting & layout is lost when creating the source document for PII processing – this is a price we pay for simplicity)
-* **context**: (optional) additional field contributing context for the chunk.
-It can contain the same three context fields as with tabular documents:
-_global_ (context defined in the document header), _before_ (previous chunk)
-and _after_ (the next chunk).
+
+## Storage format
+
+As a support file format, we define YAML files as the serialized representation
+of a Source Document. Those are easy to inspect visually and handle/edit
+manually, and also can be processed via automatic tools and packages. The YAML
+specification is actually somehow complex, but we would use only a subset of it:
+the part strictly needed to support the definitions above
+ * sequences, for the sequences of chunks
+ * mappings, for each chunk
+ * literal block scalars, to hold the text contents of each chunk
+
+YAML is therefore a good candidate for storage and for manual inspection or editing. For online APIs its equivalent representation as JSON might be more appropriate, though the result would be more involved, specially with the need to serialize the text chunks, including newlines and character escaping
+
+A serialized dump of a Source Document will contain then a YAML representation
+of the document. Its form will be a dictionary with three elements:
+ * `format`: a string identifier that signals the file as being a PII Source
+   Document. Its value is `piisa:src-document:v1`
+ * `header`: the document general metadata, as explained in [document header](#document-header)
+ * `chunks`: a list of all the chunks generated in a _structural iteration_ of
+   the document
+
+The use of structural iteration allows:
+ - the preservation of the document structure, which then can be regenerated
+   when the document is read from the file
+ - space savings by not including the context fields present only in the full
+   iteration
+
+Note that context fields that are present in the structural iteration will _be_
+dumped into the file.
+
```

Comment:
We don't talk about security, but one thing to note is that since we are serializing, the data will be stored in some medium. It would be best if it was the same medium as the original document (if from a secured database, the serializaiton is put back to the secured database). However, if the document was purely in memory, but the serialization puts it in disk, there is a greater security risk.

Reply:
I confess my ignorance on security issues. Should we mention security as an aspect "to be dealt with later", or in a parallel/wrapper standard?

Reply:
I think we can put this in the "to be dealt with later". An excercise for the reader...